Can't use type hints with _process_custom_fx

**Godot version:**
aadbb66dc23d547393bc220e1f36b56cf10f5cca

**OS/device including version:**
ArcoLinux

**Issue description:**
This function expects two things: `CustomFXChar`, a type which doesn't appear to exist (I believe it's meant to be `CharFXTransform`), and `int` return type, where, according to the PR, you're intended to return a bool.  So, the only way it works is if you leave type hints off.

`func _process_custom_fx(char_fx: CharFXTransform) -> int:`
![image](https://user-images.githubusercontent.com/28724859/64285824-f1da5980-cf21-11e9-9c00-4ab29d829718.png)

`func _process_custom_fx(char_fx: CustomFXChar) -> int:`
![image](https://user-images.githubusercontent.com/28724859/64285848-fb63c180-cf21-11e9-9615-79de4857ebaa.png)

And it generates an error when compiling with mono on Linux or cross-compiling on Linux for Windows with mono:
```Swift
 ObjectType/RichTextEffect.cs(22,45): error CS0246: The type or namespace name 'CustomFXChar' could not be found (are you missing a using directive or an assembly reference?) [/run/media/user/user/git/godot/modules/mono/glue/Managed/Generated/GodotSharp/GodotSharp.csproj]
```

--- a/scene/gui/rich_text_effect.cpp
+++ b/scene/gui/rich_text_effect.cpp
@@ -33,7 +33,7 @@
 #include "core/script_language.h"
 
 void RichTextEffect::_bind_methods() {
-	BIND_VMETHOD(MethodInfo(Variant::INT, "_process_custom_fx", PropertyInfo(Variant::OBJECT, "char_fx", PROPERTY_HINT_RESOURCE_TYPE, "CustomFXChar")));
+	BIND_VMETHOD(MethodInfo(Variant::BOOL, "_process_custom_fx", PropertyInfo(Variant::OBJECT, "char_fx", PROPERTY_HINT_RESOURCE_TYPE, "CharFXTransform")));
 }
 
 Variant RichTextEffect::get_bbcode() const {

--- a/scene/register_scene_types.cpp
+++ b/scene/register_scene_types.cpp
@@ -101,6 +101,7 @@
 #include "scene/gui/popup_menu.h"
 #include "scene/gui/progress_bar.h"
 #include "scene/gui/reference_rect.h"
+#include "scene/gui/rich_text_effect.h"
 #include "scene/gui/rich_text_label.h"
 #include "scene/gui/scroll_bar.h"
 #include "scene/gui/scroll_container.h"
@@ -344,6 +345,7 @@ void register_scene_types() {
 	ClassDB::register_class<ColorPickerButton>();
 	ClassDB::register_class<RichTextLabel>();
 	ClassDB::register_class<RichTextEffect>();
+	ClassDB::register_class<CharFXTransform>();
 	ClassDB::register_class<PopupDialog>();
 	ClassDB::register_class<WindowDialog>();
 	ClassDB::register_class<AcceptDialog>();

